evm/src/main/java/org/hyperledger/besu/evm/precompile/FalconPrecompiledContract.java

```diff
@@ -58,8 +56,7 @@ public PrecompileContractResult computePrecompile(
     Bytes methodAbi = methodInput.slice(0, METHOD_ABI.size());
     if (!methodAbi.xor(METHOD_ABI).isZero()) {
       LOG.trace("Unexpected method ABI: " + methodAbi.toHexString());
-      return PrecompileContractResult.halt(
-          null, Optional.of(ExceptionalHaltReason.PRECOMPILE_ERROR));
+      return PrecompileContractResult.success(Bytes32.leftPad(Bytes.of(1)));
```

Comment:
Why are you using `PrecompiledResult#success` instead of `PrecompiledResult#halt` here? I think it would be good for the user to distinguish between an error and a verification failure

Reply:
@diega you are right, I've added a precompile error instead of returning precompiled success.